### DISCUSSION TITLE
Binary collision model

### DISCUSSION THREAD

**Discussion by pordyna** [BLAST-WarpX/warpx#4670] [Q&A] [answered]
Hi everyone! 

I have a quick question regarding your binary Coulomb collision implementation. Does it include the corrections suggested in the 2020 Higginson paper[1] or is based only on the 2012 Perez publication[2]? The 2020 paper showed some problems with the handling of differently weighted particles in the 2012 model and introduced some changes. One of them being switching to rejection sampling. 

[1] D. P. Higginson, I. Holod, and A. Link, A corrected method for Coulomb scattering in arbitrarily weighted particle-in-cell plasma simulations, Journal of Computational Physics 413, 109450 (2020).
[2] F. Pérez, L. Gremillet, A. Decoster, M. Drouin, and E. Lefebvre, Improved modeling of relativistic collisions and collisional ionization in particle-in-cell codes, Physics of Plasmas 19, 083104 (2012).

**Comment by RemiLehe** [comment, marked as answer]
Thanks for the question!
The current WarpX implementation uses the algorithm from [2], and the Higginson correction has not been implemented yet. @Yin-YinjianZhao started implementing this correction in WarpX here: https://github.com/ECP-WarpX/WarpX/pull/3494, but for some reason our automated tests were failing with the new implementation, and @Yin-YinjianZhao has not had time to investigate this further. We should pick this up again once someone in the WarpX team has the bandwidth to do so.

**Reply by pordyna** [reply]
Thanks for the answer and good to know. In case this is of any help, I wrote a somewhat extensive explanation of the algorithm and tests  used for the implementation of [1] in PIConGPU that can be found [here](https://picongpu.readthedocs.io/en/latest/models/binary_collisions.html).
